### ISSUE TITLE
Path to arangojs 7.0.0

### ISSUE THREAD

**Issue by pluma** [arangodb/arangojs#660]
This issue serves to document the process of wrapping up the arangojs 7.0.0 release.

1. The `master` branch has been frozen.
2. Any further development of v6 until v7.0.0 is released will happen in the `v6` branch.
3. The arangojs 7 PR has been merged into the `v7` branch.
4. Any preview releases prior to v7.0.0 will happen in the `v7` branch.

Once v7.0.0 is ready to be released:

1. A final minor release will be published from the `v6` branch containing deprecations and upgrade paths where possible.
2. The `master` branch will be fast-forwarded onto the `v6` branch.
3. The `v7` branch will be rebased onto the `master` branch and git tags for the preview releases adjusted accordingly.
4. The `master` branch will then be fast-forwarded onto the `v7` branch and unfrozen.
5. The `v6` and `v7` branches will be removed.

TODO for v7.0.0:

* Inline all remaining documentation
* Check for any stray TODOs
* Add workflow for generating & publishing typedoc docs via gh-pages

**Comment by pluma** [comment]
Preview release 7.0.0-preview.0 has been released on npm with the `next` tag:

https://github.com/arangodb/arangojs/releases/tag/v7.0.0-preview.0

**Comment by pluma** [comment]
The v7 branch documentation is now continuously released at http://arangodb.github.io/arangojs/

The workflow probably needs a filter to ensure it only deploys the documentation if the commit message indicates a release (i.e. message is "x.y.z" where x, y and z are numbers).

**Comment by pluma** [comment]
The hosted documentation now contains multiple versions. `/devel` will always be updated whenever the active branch (currently `v7`) changes, `/6.x.y` are manually updated, `/7.x.y` onward are automatically published whenever a new version tag is pushed.

**Comment by pluma** [comment]
v7.0.0-preview.1 has been released today.

Changes: https://github.com/arangodb/arangojs/releases/tag/v7.0.0-preview.1
Documentation: http://arangodb.github.io/arangojs/7.0.0-preview.1/index.html

**Comment by iMoses** [comment]
I went over the changes and have a question.
why was everything related to `arangoSearchView` was changed to simply `view` except for `createArangoSearchView`?
seems to me like it should also be changed to `createView`.

Not sure this is the right place to comment this, but we are talking about 7.0.0 :)

**Comment by pluma** [comment]
Good question. Right now ArangoDB only has one View type but in the future it may have more. The ArangoSearchView is thus a kind of View like EdgeCollection is a type of collection. As there is a `createEdgeCollection` method, there's also a `createArangoSearchView` method. There's no meaningful default type, so there's no generic `createView` method (yet).

**Comment by ZhangJian-3ti** [comment]
Then createSearchView is a better name.

**Comment by ZhangJian-3ti** [comment]
Or createView with a type config, defaults as search view.

**Comment by pluma** [comment]
There's no clear use case for a future alternative View type yet, so I think it's okay to rename the method to `createView` for now even if it means requiring a breaking change in the distant future.

FWIW the view type is called `arangosearch` within ArangoDB, so `searchView` is not an option. It seems to be the only case in which the type uses "arango" as part of its name, so this is easy to stumble over.

**Comment by pluma** [comment]
Deprecations are not really an option given the most significant changes are to the collections API, so there will not be a final minor release for v6 to add these.

Right now the plan is as follows:

1. Inline the remaining documentation (i.e. Graph API).
2. Resolve any remaining `TODO`s in the code.
3. Cherry-pick any changes from `v6` into `v7`.
4. Resolve #673.
5. Rename `v7` to `main`.
6. Release `7.0.0-rc.0`.
7. Check for any currently undocumented types and values and document them as necessary.
8. Clean up the v7 CHANGELOG.
9. Publish `7.0.0` final.
10. Write a migration guide.

I hope we can get to step 6 by the end of this week or the middle of next week (i.e. 2020-07-15).

**Comment by pluma** [comment]
We've just released `v7.0.0-rc.0`, which is the final preview release before `v7.0.0` proper.

**Comment by pluma** [comment]
Okay, `v7.0.0-rc.2` will now be the final preview release, i.e. there will be no further changes before `v7.0.0`. Th only thing left to do prior to release is cleaning up the CHANGELOG.
